index.js

```diff
@@ -282,36 +328,118 @@ class GhostInspector {
     return [data, passing]
   }
 
+  async waitForResult (resultId, options = { pollInterval: DEFAULT_POLL_INTERVAL }, callback) {
+    if (typeof options === 'function') {
+      callback = options
+      options = { pollInterval: DEFAULT_POLL_INTERVAL }
+    }
+    let result
+    try {
+      let passing = null
+      while (passing === null) {
+        await wait(options.pollInterval)
+        let pollFunction = this.getTestResult
+        if (options.resultType === 'suite') {
+          pollFunction = this.getSuiteResult
+        }
+        result = await pollFunction.call(this, resultId)
```

Comment:
What about having the caller methods (`waitForSuiteResult`, `waitForTestResult`) define this logic? So instead of checking the `options` object for `resultType` and passing through the `resultId`, just give this method a poll function. For example:
```js
async waitForSuiteResult (suiteResultId, options, callback) {
    const pollFunction = () => this.getSuiteResult(suiteResultId)
    return this.waitForResult(pollFunction, callback)
}
```

Reply:
Great suggestion :)